vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridLazyDataViewImpl.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.grid.dataview;
+
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.data.provider.AbstractDataView;
+import com.vaadin.flow.data.provider.BackEndDataProvider;
+import com.vaadin.flow.data.provider.CallbackDataProvider;
+import com.vaadin.flow.data.provider.DataCommunicator;
+import com.vaadin.flow.data.provider.SizeEstimateCallback;
+
+/**
+ * Implementation of the lazy data view for grid.
+ * 
+ * @param <T>
+ *            the type of the items in grid
+ */
+public class GridLazyDataViewImpl<T> extends AbstractDataView<T>
+        implements GridLazyDataView<T> {
+
+    private final DataCommunicator<T> dataCommunicator;
+
+    /**
+     * Creates a new instance and verifies the passed data provider is
+     * compatible with this data view implementation.
+     * 
+     * @param dataCommunicator
+     *            the data communicator of the component
+     * @param component
+     *            the grid component
+     */
+    public GridLazyDataViewImpl(DataCommunicator<T> dataCommunicator,
+            Grid<T> component) {
+        super(dataCommunicator::getDataProvider, component);
+        this.dataCommunicator = dataCommunicator;
+    }
+
+    private DataCommunicator<T> getDataCommunicator() {
```

Comment:
`DataCommunicator` is going to be used for all lazy data view implementations, isn't it? If so, we could create AbstractLazyDataView and put the common methods there.

Reply:
Yes like I said already earlier, should be done. It is now there